demos/projects/ST/b-l475e-iot01a/config/demo_config.h

```diff
@@ -0,0 +1,187 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#ifndef DEMO_CONFIG_H
+#define DEMO_CONFIG_H
+
+/* FreeRTOS config include. */
+#include "FreeRTOSConfig.h"
+
+/**************************************************/
+/******* DO NOT CHANGE the following order ********/
+/**************************************************/
+
+/* Include logging header files and define logging macros in the following order:
+ * 1. Include the header file "logging_levels.h".
+ * 2. Define the LIBRARY_LOG_NAME and LIBRARY_LOG_LEVEL macros depending on
+ * the logging configuration for DEMO.
+ * 3. Include the header file "logging_stack.h", if logging is enabled for DEMO.
+ */
+
+#include "logging_levels.h"
+
+/* Logging configuration for the Demo. */
+#ifndef LIBRARY_LOG_NAME
+    #define LIBRARY_LOG_NAME    "AzureIoTDemo"
+#endif
+
+#ifndef LIBRARY_LOG_LEVEL
+    #define LIBRARY_LOG_LEVEL    LOG_INFO
+#endif
+
+/* Prototype for the function used to print to console on Windows simulator
+ * of FreeRTOS.
+ * The function prints to the console before the network is connected;
+ * then a UDP port after the network has connected. */
+extern void vLoggingPrintf( const char * pcFormatString,
+                            ... );
+
+/* Map the SdkLog macro to the logging function to enable logging
+ * on Windows simulator. */
+#ifndef SdkLog
+    #define SdkLog( message )    vLoggingPrintf message
+#endif
+
+#include "logging_stack.h"
+
+/************ End of logging configuration ****************/
+
+/**
+ * @brief Enable Device Provisioning
+ * 
+ * @note To disable Device Provisioning undef this macro
+ *
+ */
+#define democonfigENABLE_DPS_SAMPLE
+
+#ifdef democonfigENABLE_DPS_SAMPLE
+
+/**
+ * @brief Provisioning service endpoint.
+ *
+ * @note https://docs.microsoft.com/en-us/azure/iot-dps/concepts-service#service-operations-endpoint
+ * 
+ */
+#define democonfigENDPOINT                  "global.azure-devices-provisioning.net"
+
+/**
+ * @brief Id scope of provisioning service.
+ * 
+ * @note https://docs.microsoft.com/en-us/azure/iot-dps/concepts-service#id-scope
+ * 
+ */
+#define democonfigID_SCOPE                  "<YOUR ID SCOPE HERE>"
```

Comment:
General feedback here and other places like Linux simulator.  If I'm a customer coming to bring up my board, despite trimming as much as we can there's still a fair amount of knobs I need to set to even connect to Azure IoT Hub/DPS in 1st place, never mind chosing between them in 1st place along with SaS/x509 and all that.

Beyond just documentation, could we have some sort of convention like
// TODO: Fill this in

So that someone could at least Control-F through a file to see what was required versus the stuff they can tweak if they want but probably won't on initial bring up, like they probably don't care about mucking with stack sizes just to see bytes flowing to Azure. 

There's probably more refinement needed here, since not all these fields are required for every permutation (in fact DPS versus hub has mutually exclusive stuff.)

Reply:
I want us to eventually support KConfig for this very reason. Menu system that allows for Configurations variables to be accessed and set via this menuconfig systems and variables are access across both CMake and in the code prefixed with CONFIG. You would then get to see what the values are and would be able to change them and behavior without going into code. ESP32 IDF and Zephyr used this by default but most folks just add Kconfig into their system for STM32 and other uCs. Now that we have a great CMake system this would be a great next step. -TEW

Reply:
updated with todo.